packages/react-ui-core/src/Gmap/GmapSpinner.js

```diff
@@ -11,18 +11,44 @@ export default class MapSpinner extends PureComponent {
     loading: PropTypes.bool,
     theme: PropTypes.object,
     className: PropTypes.string,
+    closeDelay: PropTypes.number,
   }
 
   static defaultProps = {
+    closeDelay: 0,
     loading: false,
     theme: {},
     color: '#000',
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: this.props.loading,
+    }
+  }
+
+  componentDidUpdate(nextProps) {
```

Comment:
did you mean to use a difference lifecycle method here, because 

`componentDidUpdate(prevProps, prevState, snapshot)`

Reply:
Here, no. @mneumark asked the same.  Reason being is because `componentWIllReceiveProps` will become deprecated and `getDerivedStateFromProps` won't work how we need it to with a `setTmeout`. I'm open to changing this to something else though...

Reply:
I think calling close in `componentDidUpdate` is the right thing to do, just calling the param `nextProps` is weird/unexpected.